lib/usd/hdMaya/adapters/meshAdapter.cpp

```diff
@@ -0,0 +1,444 @@
+//
+// Copyright 2019 Luma Pictures
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include <pxr/pxr.h>
+
+#include <pxr/base/tf/type.h>
+
+#include <pxr/usd/usdGeom/tokens.h>
+
+#include <pxr/imaging/hd/tokens.h>
+#include <pxr/imaging/pxOsd/tokens.h>
+
+#include <maya/MAnimControl.h>
+#include <maya/MCallbackIdArray.h>
+#include <maya/MDGContext.h>
+#include <maya/MDGContextGuard.h>
+#include <maya/MFloatArray.h>
+#include <maya/MFnMesh.h>
+#include <maya/MIntArray.h>
+#include <maya/MItMeshPolygon.h>
+#include <maya/MNodeMessage.h>
+#include <maya/MObjectHandle.h>
+#include <maya/MPlug.h>
+#include <maya/MPolyMessage.h>
+
+#include "adapterDebugCodes.h"
+#include "adapterRegistry.h"
+#include "mayaAttrs.h"
+#include "shapeAdapter.h"
+#include "tokens.h"
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+namespace {
+
+constexpr int MAX_SMOOTH_LEVEL = 8;
+
+const std::array<std::pair<MObject&, HdDirtyBits>, 8> _dirtyBits{
```

Comment:
Another 8? https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-magic

Reply:
This one isn't magic - std::array requires you to explicitly state how many members are in it at creation - this is essentially the equivalent of:

int myInts[8] = {1, 2, 3, 4, 5, 6, 7, 8};

... unfortunately, unlike a normal array, I don't know of a good way to have it fill it in automatically, based on the initializer.

I suppose we might be able to convert to a normal array? I'll check on that in my next round of updates...

Reply:
Didn't see anything in the code which was using interfaces particular to std::array, so I just converted to a "normal" `myArray[]` style array:

https://github.com/Autodesk/maya-usd/pull/231/commits/d4266efa8f10b43c463e78417c1b9bccfd33b6f4

Reply:
Thank you.